### DISCUSSION TITLE
`usePrepareContractWrite` returning `config.request?.gas` of `undefined`

### DISCUSSION THREAD

**Discussion by Ncookiez** [wevm/wagmi#2988] [Q&A]
Hey guys! Was looking to pad the gas estimate a little with something similar to [#1129](https://github.com/wagmi-dev/wagmi/discussions/1129#discussioncomment-3957268). Am aware `gasLimit` has been renamed to `gas` a little while back.

I am not seeing a gas estimation being returned though, as logging `config.request?.gas` always returns `undefined`.

Am using `usePrepareContractWrite` as follows:

```ts
const { config } = usePrepareContractWrite({
  chainId: vault.chainId,
  address: vault.address,
  abi: vaultABI,
  functionName: 'deposit',
  args: [amount, userAddress],
  enabled
})

console.log('🍪 ~ config gas:', config.request?.gas)
```

This is a somewhat standard ERC4626 vault, and the `enabled` var is just checking for valid params, allowance checks, etc.

Am I missing something here? The transaction itself always works as intended, but sometimes the gas limit is reached which is what I'm attempting to remedy. Thanks in advance!

**Comment by Ncookiez** [comment]
For added context after further testing: Viem's `publicClient.estimateContractGas` function works as intended and returns an accurate gas amount estimate.

The only difference in params between the two being that I am passing `account` into `estimateContractGas`, but not `usePrepareContractWrite`. Adding the `account` param to `usePrepareContractWrite` unfortunately still does not get me a `gas` result.

**Reply by Yuripetusko** [reply]
Not sure, but I would try and pass `walletClient` (previously signer) to it

**Comment by Zimtente** [comment]
also not working for me. manually estimating gas works fine but the prepareData always returns undefined for me too

**Comment by Ncookiez** [comment]
Is is possible the docs description of "Eagerly fetches the parameters required for sending a contract write transaction such as the gas estimate." [here](https://wagmi.sh/react/prepare-hooks/usePrepareContractWrite) is just misleading?

Does gas estimation through the `usePrepareContractWrite` hook work for anyone?
